core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -499,4 +501,47 @@ public DataMapFactory getDataMapFactory() {
     }
     return prunedSegments;
   }
+
+  /**
+   * Prune the datamap of the given segments and return the Map of blocklet path and row count
+   *
+   * @param segments
+   * @param partitions
+   * @return
+   * @throws IOException
+   */
+  public Map<String, Long> getBlockRowCount(List<Segment> segments,
+      final List<PartitionSpec> partitions, TableDataMap defaultDataMap)
+      throws IOException {
+    Map<String, Long> blockletToRowCountMap = new HashMap<>();
+    for (Segment segment : segments) {
+      List<CoarseGrainDataMap> dataMaps = defaultDataMap.getDataMapFactory().getDataMaps(segment);
+      for (CoarseGrainDataMap dataMap : dataMaps) {
+        dataMap.getRowCountForEachBlock(segment, partitions, blockletToRowCountMap);
+      }
+    }
+    return blockletToRowCountMap;
+  }
+
+  /**
+   * Prune the datamap of the given segments and return the Map of blocklet path and row count
+   *
+   * @param segments
+   * @param partitions
+   * @return
+   * @throws IOException
+   */
+  public Map<String, Long> getRowCount(List<Segment> segments,
```

Comment:
It should just return long, not Map.

Reply:
This map will have only one entry. We need a map to form BlockMappingVO object which in turn is used in CarbonUpdateUtil#getRowCount method(called from CarbonCountStar), where we check for the IUD scenario.

Reply:
I am aware of IUD scenario but  I am not sure why IUD scenario related to datamap in this API, If your API says `getRowCount` then just return the row count as long , not the Map. Caller methods should handle the IUD scenarios, not in this API.

Reply:
ok. done